### PR TITLE
SDK - Fix SDK on Python 3.8

### PR DESCRIPTION
**Pull request by Ark-kun**
Fixes the follwoing error: "TypeError: code() takes at least 14 arguments (13 given)".

The cause of the issue is a breaking change in CodeType constructor in Python 3.8.
https://bugs.python.org/issue37221
This should have been fixed by https://github.com/python/cpython/pull/13959 and https://github.com/python/cpython/pull/14505, but the code still fails.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3126)

### CODE REVIEW

**Comment by pablogsal** [comment]
> This should have been fixed by [python/cpython#13959](https://github.com/python/cpython/pull/13959) and [python/cpython#14505](https://github.com/python/cpython/pull/14505), but the code still fails.

Those PRs did not intend to change back the `CodeType` constructor. As I said in https://github.com/python/cpython/pull/12701#discussion_r380685333:

Also, `types.CodeType`'s constructor is considered private. We now even have this warning in the docs (https://docs.python.org/3.8/library/types.html#standard-interpreter-types):

> If you instantiate any of these types, note that signatures may vary between Python versions.

Additionally, you have now available `types.CodeType.replace` in case you can to create a copy with some fields changed.

File: sdk/python/kfp/components/_dynamic.py [hunk 1]
```diff
@@ -44,21 +45,31 @@ def pass_locals():
...
-        code.co_names,
-        mod_co_varnames,
-        mod_co_filename,
-        mod_co_name,
-        mod_co_firstlineno,
-        code.co_lnotab
-    )
+    if sys.version_info >= (3, 8):
+        modified_code = (code
+            .replace(co_argcount=mod_co_argcount)
```
Comment (pablogsal) [inline sdk/python/kfp/components/_dynamic.py]: You don't need all these `replace` calls. You can simply do:

```
code.replace(co_argcount=mod_co_argcount, co_nlocals=mod_co_nlocals, co_varnames=mod_co_varnames, co_filename=mod_co_filename, co_name=mod_co_name, co_firstlineno=mod_co_firstlineno)
```

Comment (Ark-kun) [reply sdk/python/kfp/components/_dynamic.py]: Thanks!
It looks better now.

**Comment by Ark-kun** [comment]
> Hi, CPython core dev here 
> 
> Those PRs did not intend to change back the `CodeType` constructor. As I said in [python/cpython#12701 (comment)](https://github.com/python/cpython/pull/12701#discussion_r380685333):
> 
> Also, `types.CodeType`'s constructor is considered private. We now even have this warning in the docs (https://docs.python.org/3.8/library/types.html#standard-interpreter-types):
> 
> > If you instantiate any of these types, note that signatures may vary between Python versions.

Thank you for the explanation!
I hope in future we'll have a more stable API for dynamic function generation.

**Comment by Ark-kun** [comment]
@numerology Can you please take a look?
/approve

**Pull request merged**